learn/tooling-guide/ballerina-shell.md

```diff
@@ -2,11 +2,197 @@
+
+## Starting the Shell
+
+Execute the command below to start Ballerina Shell.
+
+```bash
+$ bal shell
+Welcome to Ballerina Shell REPL.
+Type /exit to exit and /help to list available commands.
+
+=$ 
+```
+
+## Stopping the Shell
+
+Execute the `/exit` command below in a new prompt to exit the Shell.
+
+```bash
+=$ /exit                                                                                                                                                                                                           
+| Bye!!!
+```
+
+## Evaluating Code Snippets
+
+The types of code snippets below are supported in the Shell.
+
+- Import declarations
+- Module-level declarations (except services)
+- Statements
+- Expressions
+
+### Importing a Module
+
+A module can be imported as follows. Once a module is imported, it can be used in any subsequent snippet run in that particular Shell session. This is true for other module-level declarations as well.
+
+```bash
+=$ import ballerina/io
+```
+
+Execute the `/imports` command below to view the list of the imported modules.
+
+```bash
+=$ /imports
+| ('io) import ballerina/'io as 'io;
+```
+
+### Executing Statements
+
+Any valid Ballerina statement can be evaluated in the Shell. The example below evaluates a variable declaration statement.
+
+```bash
+=$ string name = "Ballerina Shell" 
+```
+
+
+### Evaluating Expressions
+
+An expression always produces a value when evaluated. You can use the Shell to evaluate any arbitrary expression. It will output the result of the evaluation.
+
+```bash
+=$ 10 + 20 * 5                                                                                                                                                                                                       
+110
+```
+
+A reference to a variable is also an expression. Thus, evaluating the name of the previously-defined variable will return its value.
+
+```bash
+=$ name                                                                                                                                                                                                            
+"Ballerina Shell"
+```
+
+###  Defining Types
+
+Type definitions can be written just as you write them in a source file. The Shell supports multi-line snippets.
+
+```bash
+=$ type Person record {                                                                                                                                                                                              
+ > string name;
+ > int age;
+ > }
+```
+
+Execute the `/dclns` command below to view the defined types (and other module-level declarations).
```

Comment:
```suggestion
Execute the `/dclns` command to view the defined types (and other module-level declarations).
```

Reply:
Same comment as above.

Reply:
This is a different case. This is not referring to the code snippet. The code snippet is there just as an illustrative example of what's described here.

Reply:
Got it. Then, shall we say: "Execute the `/dclns` command to view the defined types (and other module-level declarations) as shown below."?